pwiz_tools/Skyline/TestPerf/DetectionsPlotTest.cs

```diff
@@ -0,0 +1,215 @@
+                "Cache is not updated on document change.");
+
+            //verify that the cache is purged after the document update
+            RunUI(() =>
+            {
+                Assert.IsTrue(DetectionPlotData.DataCache.Datas.All((dat) =>
+                    ReferenceEquals(SkylineWindow.DocumentUI, dat.Document)));
+            });
+            AssertDataCorrect(pane, 4, 0.001f);
+
+            Trace.WriteLine(this.GetType().Name + ": showing a histogram pane.");
+            RunUI(() => { SkylineWindow.ShowDetectionsHistogramGraph(); });
+            WaitForGraphs();
+            DetectionsHistogramPane paneHistogram;
+            var graphHistogram = SkylineWindow.DetectionsPlot;
+            Assert.IsTrue(graphHistogram.TryGetGraphPane(out paneHistogram), "Cannot get histogram pane.");
+            //display and hide tooltip
+            evt = FindBarPoint(paneHistogram, graphHistogram, 5);
+            Trace.WriteLine(this.GetType().Name + ": showing histogram tooltip.");
+            RunUI(() =>
+            {
+                paneHistogram.HandleMouseMoveEvent(graphHistogram.GraphControl, evt);
+                Assert.IsNotNull(paneHistogram.ToolTip, "No tooltip found.");
+                WaitForConditionUI(() => paneHistogram.ToolTip.IsVisible, "Tooltip is not visible.");
+                //verify the tooltip text
+                Assert.IsTrue(HISTOGRAM_TIP_TEXT.SequenceEqual(paneHistogram.ToolTip.TipLines));
+                paneHistogram.HandleMouseMoveEvent(graphHistogram.GraphControl, new MouseEventArgs(MouseButtons.None, 0, 0, 0, 0));
+                Assert.IsFalse(paneHistogram.ToolTip.IsVisible);
+            });
+            RunUI(() =>
+            {
+                graph.Close();
+                graphHistogram.Close();
+            });
+            WaitForGraphs();
+            Trace.WriteLine(this.GetType().Name + ": Test complete.");
+        }
+
+        private void AssertDataCorrect(DetectionsPlotPane pane, int refIndex, float qValue, bool record = false)
+        {
+            DetectionPlotData data = null;
+            Trace.WriteLine(this.GetType().Name + $": Waiting for data for qValue {qValue} .");
+            WaitForConditionUI(() => (data = pane.CurrentData) != null 
+                                           && pane.CurrentData.QValueCutoff == qValue
+                                           && DetectionPlotData.DataCache.Status == DetectionPlotData.DetectionDataCache.CacheStatus.idle,
+                () => $"Retrieving data for qValue {qValue}, refIndex {refIndex} took too long.");
+            WaitForGraphs();
+            Assert.IsTrue(data.IsValid);
+
+            if (record)
+            {
+                Debug.WriteLine("Peptides");
+                pane.CurrentData.GetTargetData(DetectionsGraphController.TargetType.PEPTIDE).TargetsCount
+                    .ForEach((cnt) => { Debug.Write($"{cnt}, "); });

```

Comment:
We usually use "Console.WriteLine" for things (such as record mode output) that we actually want the developer to see.

Debug.Write gets completely stripped out in the Release build (which is how we usually run tests).